include/envoy/stream_info/stream_info.h

```diff
@@ -578,6 +578,16 @@ class StreamInfo {
    * @param id Connection ID of the downstream connection.
    **/
   virtual void setConnectionID(uint64_t id) PURE;
+
+  /**
+   * @param filter_chain_name Network filter chain name of the downstream connection.
+   */
+  virtual void setFilterChainName(absl::string_view filter_chain_name) PURE;
+
+  /**
+   * @return Network filter chain name of the downstream connection.
+   */
+  virtual const std::string& filterChainName() const PURE;
```

Comment:
Why return "const std::string&" instead of absl::string_view or std::string ?

Reply:
I am following the convention here.
https://github.com/envoyproxy/envoy/pull/14983/files/385867984d9ffb954b025e2c24e87947fb61d7be#diff-ab3ff2fdafab752822b91b1a7f7d48d90d4e308e074be16c6a7bc00ab9d5fb8bR526

I can sanitize the entire file in the next PR

Reply:
that link is not working well for me, what convention are you referring to?

Reply:
That is the signature 60 lines above: 
```
  virtual const std::string& requestedServerName() const PURE;
```